### PR TITLE
rocm packages: Update the version for 5.3.0 release

### PR DESCRIPTION
**Pull request by srekolam**
This PR contains the version updates for some of the rocm recipes for rocm-5.3.0 release. This PR is dependent on PR #33320 .

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Confirmed the sha256 for each of the 8 packages' version directives.

**Review by cgmb — DISMISSED** [review]

**Review by renjithravindrankannath — CHANGES_REQUESTED** [review]
Need -DCMAKE_INSTALL_LIBDIR=lib in hipsolver and hipfft 

```
        if self.spec.satisfies("@5.3.0:"):
            args.append("-DCMAKE_INSTALL_LIBDIR=lib")
```

File: var/spack/repos/builtin/packages/hipsolver/package.py [hunk 1]
```diff
@@ -84,9 +95,12 @@ def cmake_args(self):
         args = [
             self.define("BUILD_CLIENTS_SAMPLES", "OFF"),
             self.define("BUILD_CLIENTS_TESTS", self.run_tests),
+            self.define("BUILD_CLIENTS_TESTS", self.run_tests),
```
Comment (cgmb) [inline var/spack/repos/builtin/packages/hipsolver/package.py]: There appears to be a duplicate line introduced here.

File: var/spack/repos/builtin/packages/rocm-openmp-extras/package.py [hunk 2]
```diff
@@ -246,10 +254,14 @@ def setup_run_environment(self, env):
         devlibs_prefix = self.spec["llvm-amdgpu"].prefix
         openmp_extras_prefix = self.spec["rocm-openmp-extras"].prefix
         llvm_prefix = self.spec["llvm-amdgpu"].prefix
+        hsa_prefix = self.spec["hsa-rocr-dev"].prefix
         env.set("AOMP", "{0}".format(llvm_prefix))
         env.set("HIP_DEVICE_LIB_PATH", "{0}/amdgcn/bitcode".format(devlibs_prefix))
         env.prepend_path("CPATH", "{0}/include".format(openmp_extras_prefix))
         env.prepend_path("LIBRARY_PATH", "{0}/lib".format(openmp_extras_prefix))
+        if self.spec.version >= Version("5.3.0"):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/rocm-openmp-extras/package.py]: We generally prefer use of `self.spec.satisfies("5.3.0:")` for version range checks.  See https://spack.readthedocs.io/en/latest/packaging_guide.html#testing-spec-constraints.

**Comment by renjithravindrankannath** [comment]
@spackbot run pipeline

**Review by cgmb — DISMISSED** [review]

**Comment by cgmb** [comment]
@spackbot run pipeline

**Comment by srekolam** [comment]
@spackbot run pipeline

**Comment by srekolam** [comment]
@cgmb  seeing an failure with rocprim@5.3.0 for petsc in gitlab-ci , any clues ?
https://gitlab.spack.io/spack/spack/-/jobs/4980002

**Comment by cgmb** [comment]
> @cgmb seeing an failure with rocprim@5.3.0 for petsc in gitlab-ci , any clues ? https://gitlab.spack.io/spack/spack/-/jobs/4980002

I'm rather surprised, since this PR doesn't change rocthrust or rocprim.

**Comment by cgmb** [comment]
To me, that looks like a legitimate bug in one of rocprim, rocthrust or petsc. It's triggered by this template instantiation:

    PetscCallThrust(thrust::merge(thrust::device, Azb, Aze, Bzb, Bze, Czb, IJCompare4()));

The key failure seems to be:

```
include/rocprim/device/detail/device_merge.hpp:77:9: note: candidate template ignored: 
 deduced conflicting types for parameter 'KeysInputIterator' ('zip_iterator<tuple<[...], thrust::detail::normal_iterator<thrust::device_ptr<int>>, [8 * ...]>>' vs. 'zip_iterator<tuple<[...],
  thrust::transform_iterator<Shift, thrust::detail::normal_iterator<thrust::device_ptr<int>>>, [8 * ...]>>')
2103    OffsetT merge_path(KeysInputIterator keys_input1,
```

The failure is ultimately in [the instantiation of merge_path in `partition_kernel_impl`](https://github.com/ROCmSoftwarePlatform/rocPRIM/blob/rocm-5.3.0/rocprim/include/rocprim/device/detail/device_merge.hpp#L133-L140), but I assume that's because it was passed invalid arguments from somewhere else. That said... I'm certainly no expert. This is just my best guess.

**Comment by srekolam** [comment]
@renjithravindrankannath , @cgmb @tldahlgren  can you please approve this PR.

**Review by renjithravindrankannath — APPROVED** [review]
LGTM

**Review by cgmb — APPROVED** [review]
This seems like a reasonable approach to addressing the problem. Ideally, we would have added rocprim@5.3.3 rather than patching rocprim@5.3.0, but that would be rather difficult given how all the rocm packages specify exact version dependencies.

**Comment by srekolam** [comment]
@tldahlgren can you please review and let me know if this can be merged.
Thanks!!

**Comment by srekolam** [comment]
@tldahlgren can you please review /approve this PR as this contains the changes for rocm-5.3.0 changes. This will complete the activity for rocm-5.3.0 release.

**Review by tldahlgren — APPROVED** [review]
Re-confirmed the `version()` sha256.  Rest LGTM.

**Pull request merged**